kolena/_api/v2/_metric.py

```diff
@@ -0,0 +1,302 @@
+# Copyright 2021-2024 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+from typing import Literal
+from typing import Optional
+from typing import Union
+
+from pydantic import Field
+from pydantic import field_validator
+from pydantic.dataclasses import dataclass
+from typing_extensions import Annotated
+
+MetricFormat = Literal["default", "integer", "decimal", "percentage", "scientific", "dollars", "euros"]
+
+
+@dataclass(frozen=True, order=True)
+class Highlight:
+    higherIsBetter: Union[bool, None] = None
+
+
+@dataclass(frozen=True)
+class BaseMetric:
+    label: str
+    source: Literal["result"]
+
+
+AveragingMethod = Literal["macro", "micro", "weighted"]
+ClassifierAggregator = Literal["precision", "recall", "f1_score"]
+SimpleAggregator = Literal["count", "min", "max", "mean", "median", "stddev", "sum"]
+Source = Literal["datapoint", "result"]
+
+
+@dataclass(frozen=True)
+class SimpleParams:
+    key: str
+
+
+@dataclass(frozen=True)
+class SimpleMetric(BaseMetric):
+    aggregator: SimpleAggregator
+    params: SimpleParams
+    format: Union[MetricFormat, None] = None
+    highlight: Union[Highlight, None] = None
+
+
+@dataclass(frozen=True)
+class CategoricalAggregator:
+    aggregator: Literal["count", "rate"]
+    fieldValue: Union[bool, str]
+
+
+@dataclass(frozen=True)
+class CategoricalMetric(BaseMetric):
+    aggregator: CategoricalAggregator
+    params: SimpleParams
+    format: Union[MetricFormat, None] = None
+    highlight: Union[Highlight, None] = None
+
+
+@dataclass(frozen=True, order=True)
+class BinaryClassificationAggregator:
+    aggregator: Union[Literal["accuracy"], Literal["auc"], ClassifierAggregator]
+    type: Literal["binary-classification"]
+
+
+@dataclass(frozen=True, order=True)
+class MulticlassClassificationAggregator:
+    aggregator: Union[Literal["accuracy"], ClassifierAggregator]
+    averagingMethod: Annotated[Union[AveragingMethod, None], Field(validate_default=True)] = None
+
+    @field_validator("averagingMethod")
+    @classmethod
+    def set_null_averaging_method(cls, v: Union[str, None]) -> str:
+        return v or "macro"
+
+
+@dataclass(frozen=True, order=True)
+class RegressionAggregator:
+    aggregator: Literal["mae", "mse", "rmse", "r_squared", "pearson_corr", "spearman_corr"]
+
+
+@dataclass(frozen=True, order=True)
+class ClassificationParams:
```

Comment:
How did we end up with a mix of camelCase and snake_case? It's sloppy to have this in our API.

Reply:
Not sure if there's anything to do about this -- do we need all of these classes though? Really annoying to make this public.

Reply:
I just copied everything from the server side, We need the QS and Testing endpoint to function properly which involves all the complex Filter models. We could simplify this to strictly what we need and and the additional stuff later, but I don't think this is removing too much of the models as there can be filters in stratifications of a qs.

Reply:
Yeah, this is frustrating, but technically still non-public under `_api` -- just frustrating, but not worth changing.